### PR TITLE
Added try/catch for IO errors that are logged to log4net. This preven…

### PR DESCRIPTION
**Pull request by bvadala**
…ts unhandled exceptions which can crash the process

### CODE REVIEW

**Comment by dlbroadfoot** [comment]
+1

**Review by dlbroadfoot — APPROVED** [review]

File: src/XDMessaging.Lite/Transport/IOStream/XDIOStreamBroadcaster.cs [hunk 1]
```diff
@@ -136,7 +139,15 @@ private void CleanUpMessages(object state)
                     {
                     }
                     CleanUpMessages(directory, messageTimeoutInMilliseconds);
-                    mutex.ReleaseMutex();
+
+                    try
+                    { 
+                        mutex.ReleaseMutex();
```
Comment (soleon) [inline src/XDMessaging.Lite/Transport/IOStream/XDIOStreamBroadcaster.cs:145]: Why mute this line too? I didn't know this was causing the crash.

Comment (bvadala) [reply src/XDMessaging.Lite/Transport/IOStream/XDIOStreamBroadcaster.cs:145]: you sent me the stack trace from a crash:

```
System.ApplicationException: Object synchronization method was called from an unsynchronized block of code.
   at System.Threading.Mutex.ReleaseMutex()
   at XDMessaging.Transport.IOStream.XDIOStreamBroadcaster.CleanUpMessages(Object state)
```

Comment (soleon) [reply src/XDMessaging.Lite/Transport/IOStream/XDIOStreamBroadcaster.cs:145]: Ah forgot about this one.

**Pull request merged**